controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -78,53 +74,29 @@ func (r *DataScienceClusterReconciler) Reconcile(ctx context.Context, req ctrl.R
 	log.Info("Reconciling DataScienceCluster resources", "Request.Name", req.Name)
 
 	instance := &dscv1.DataScienceCluster{}
-	err := r.Client.Get(ctx, req.NamespacedName, instance)
```

Comment:
this `Get` is still required to get the actual `DataScienceCluster` instance on the cluster

Reply:
Moving this to DSC controller causes throws an unused/ineffectual assignment error.

Reply:
you also need to check the error, but if not done, then you won't get the actual instance

Reply:
So even after i add this statement back, I'm not able to get the instance. It goes into the wrong case(DSC nonexist case when DSC exists) and returns `instance.Name` as empty.